tools/scalajsenv.js

```diff
@@ -864,14 +881,21 @@ initArray(
   const componentBase = componentData.arrayBase || componentData;
   const arrayDepth = componentData.arrayDepth + 1;
 
+  const typeTag = (((1 << 8) | arrayDepth) << 23) | componentBase.typeTag;
+
+  ArrayClass.prototype.$typeTag = typeTag;
+  ScalaJS.ClassData[typeTag] = this;
```

Comment:
Could we do away without this line? In other words, can we avoid having to store the `TypeData` of array types in `ScalaJS.ClassData`?
My concern is that this will turn `ClassTag` into a sparse array, considerably hurting its performance (access is not constant-time anymore).